browser/brave_wallet/brave_wallet_service_factory.cc

```diff
@@ -53,13 +55,18 @@ BraveWalletServiceFactory::~BraveWalletServiceFactory() = default;
 
 KeyedService* BraveWalletServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* context) const {
-  return new BraveWalletService(
+  auto* wallet_service = new BraveWalletService(
 #if defined(OS_ANDROID)
       std::make_unique<BraveWalletImporterDelegate>(),
 #else
       std::make_unique<BraveWalletImporterDelegateImpl>(context),
 #endif
       user_prefs::UserPrefs::Get(context));
+
+  // g_brave_browser_process may be null in unit tests.
+  if (g_brave_browser_process)
+    g_brave_browser_process->brave_stats_updater()->AddObserver(wallet_service);
```

Comment:
we need to remove the observer at some point

Reply:
switched to timer, as you suggested